spec.md

```diff
@@ -2867,7 +2901,8 @@ width = "<" , int , ">" ;
 binarypoint = "<<" , int , ">>" ;
 type_ground = "Clock" | "Reset" | "AsyncReset"
             | ( "UInt" | "SInt" | "Analog" ) , [ width ]
-            | "Fixed" , [ width ] , [ binarypoint ] ;
+            | "Fixed" , [ width ] , [ binarypoint ]
+            | "String" ;
```

Comment:
can we remove this from `type_ground` so that we aren't confusing whether a `Reg` can hold a `String` and whatnot? it seems like we don't need it except for `parameter` and `printf` in this PR?